development/auto-changelog.sh

```diff
@@ -1,26 +1,62 @@
-#! /bin/bash
-# update tags
+#!/usr/bin/env bash
+
+set -e
+set -u
+set -o pipefail
+
+readonly URL='https://github.com/MetaMask/metamask-extension'
+
 git fetch --tags
-# get origin
-URL='https://github.com/MetaMask/metamask-extension'
-# get git logs from last tag until HEAD, pretty by 'subject::body' filtered by grep for PRs made with Github squash merge or Github regular merge
-LOG=$(git log $(git describe --tags $(git rev-list --tags --max-count=1))..HEAD --pretty="%s::%b" --reverse --grep="Merge pull request #" --grep="(#");
-while read -r line; do
-    # get git log subject
-    SUBJECT=$(echo $line | sed -E 's/(.*):{2}(.*)/\1/')
-    # get git log PR id, PR made with Github squash merge or Github regular merge
-    PR=$(echo $SUBJECT | sed 's/^.*(#\([^&]*\)).*/\1/' | sed 's/^.*#\([^&]*\) from.*/\1/')
-    # if PR made with Github squash merge, subject is the body
-    if [ -z "$(echo $line | sed -E 's/(.*):{2}(.*)/\2/')" ]; then
-        BODY=$(echo $SUBJECT | sed "s/(#$PR)//g"); else
-        BODY=$(echo $line | sed -E 's/(.*):{2}(.*)/\2/') 
+
+most_recent_tag="$(git describe --tags "$(git rev-list --tags --max-count=1)")"
+
+readarray -t commits <<< "$(git rev-list "${most_recent_tag}"..HEAD)"
```

Comment:
`readarray` doesn't work in Bash 3. We might be able to use this kind of setup for the loop:

```diff
@@ -10,9 +10,7 @@ git fetch --tags
 
 most_recent_tag="$(git describe --tags "$(git rev-list --tags --max-count=1)")"
 
-readarray -t commits <<< "$(git rev-list "${most_recent_tag}"..HEAD)"
-
-for commit in "${commits[@]}"
+git rev-list "${most_recent_tag}"..HEAD | while read commit
 do
     subject="$(git show -s --format="%s" "$commit")"
 
```

Reply:
😞 `readarray` is great! macOS users using default Bash have it rough.

The `while read` loop is a good suggestion though; that's better than what we had before. We don't need the array.